substrate/primitives/metadata-ir/src/types.rs

```diff
@@ -48,6 +48,8 @@ pub struct MetadataIR<T: Form = MetaForm> {
 pub struct RuntimeApiMetadataIR<T: Form = MetaForm> {
 	/// Trait name.
 	pub name: T::String,
+	/// The base api_version declared on the trait.
```

Comment:
We don't need any of these changes, if you do the filtering directly in `runtime_metadata` by forwarding there the impl api version.

Reply:
Makes sense; I can remove the `base_version` from that and rely on `mod_name::VERSION` instead!

One thing this bumps into is that `VERSION` is a u32 whereas we use u64s to track method versions (which causes a type error when comparing one with the other). Any objection to me making method_versions be treated as `u32`s too (or alternately, make `VERSION` a u64)?

Reply:
`u32` sounds reasonable.